src/plugins/discourse/createGraph.js

```diff
@@ -0,0 +1,165 @@
+// @flow
+
+import {
+  Graph,
+  NodeAddress,
+  EdgeAddress,
+  type Node,
+  type Edge,
+  type NodeAddressT,
+} from "../../core/graph";
+import {type PostId, type TopicId, type Post, type Topic} from "./fetch";
+import {type DiscourseData} from "./mirror";
+import {
+  topicNodeType,
+  postNodeType,
+  userNodeType,
+  authorsPostEdgeType,
+  authorsTopicEdgeType,
+  postRepliesEdgeType,
+  topicContainsPostEdgeType,
+} from "./declaration";
+
+export function topicAddress(serverUrl: string, id: TopicId): NodeAddressT {
+  return NodeAddress.append(topicNodeType.prefix, serverUrl, String(id));
+}
+export function postAddress(serverUrl: string, id: PostId): NodeAddressT {
+  return NodeAddress.append(postNodeType.prefix, serverUrl, String(id));
+}
+export function userAddress(serverUrl: string, username: string): NodeAddressT {
+  return NodeAddress.append(userNodeType.prefix, serverUrl, username);
+}
+
+export function userNode(serverUrl: string, username: string): Node {
+  const url = `${serverUrl}/u/${username}/`;
+  const description = `[@${username}](${url})`;
+  return {
+    address: userAddress(serverUrl, username),
+    description,
+    timestampMs: null,
+  };
+}
+
+export function topicNode(serverUrl: string, topic: Topic): Node {
+  const url = `${serverUrl}/t/${String(topic.id)}`;
+  const description = `[${topic.title}](${url})`;
+  const address = topicAddress(serverUrl, topic.id);
+  return {address, description, timestampMs: topic.timestampMs};
+}
+
+export function postNode(
+  serverUrl: string,
+  post: Post,
+  topicTitle: string
+): Node {
+  const url = `${serverUrl}/t/${String(post.topicId)}/${String(post.id)}`;
+  const descr = `[post #${post.indexWithinTopic} on ${topicTitle}](${url})`;
```

Comment:
This topic title almost certainly needs to be sanitized, right? Should
also probably be `“quoted”` just for readability.